Sources/Utility/Process.swift

```diff
@@ -292,9 +323,11 @@ public final class Process: ObjectIdentifierProtocol {
     ///
     /// Note: This will signal all processes in the process group.
     public func signal(_ signal: Int32) {
-        assert(launched, "The process is not yet launched.")
-        assert(result == nil, "Trying to signal \(signal) but the process has already terminated. \(result.debugDescription)")
-        _ = libc.kill(-processID, signal)
+        self.serialQueue.sync {
+            assert(_launched, "The process is not yet launched.")
+            assert(_result == nil, "Trying to signal \(signal) but the process has already terminated. \(_result.debugDescription)")
+            _ = libc.kill(-processID, signal)
+        }
```

Comment:
I see, the problem is waitpid will hold the lock and this method will wait until that is finished, which defeats the purpose of signal.

we can remove the `result==nil` assertion because sometimes it is possible that we end up signalling processes which terminated. @neonichu and I figured this out when I was implementing this but I forgot to remove this assertion.
Also since we're just reading `launched` here and it is not expected to mutate after `launch()`, we can remove the serial queue here. That should fix the issues with process set.

yay that tests caught it 🎉